packages/react/ds-app-launchpad/src/ui/FileTree/Provider.tsx

```diff
@@ -0,0 +1,96 @@
+import type React from "react";
+import { useCallback, useReducer } from "react";
+import Context from "./Context.js";
+import "./styles.css";
+import type {
+  FileSelectionOptions,
+  ProviderOptions,
+  SearchOptions,
+} from "./types.js";
+
+type State = {
+  internalSearchQuery: SearchOptions["searchQuery"];
+  internalSelectedFile: FileSelectionOptions["selectedFile"];
+};
+
+type Action =
+  | { type: "SEARCH"; payload: SearchOptions["searchQuery"] }
+  | { type: "SELECT_FILE"; payload: FileSelectionOptions["selectedFile"] };
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case "SEARCH":
+      return { ...state, internalSearchQuery: action.payload };
+    case "SELECT_FILE":
+      return { ...state, internalSelectedFile: action.payload };
+    default:
+      return state;
+  }
```

Comment:
Perhaps the default case should create a `console.warn` in development environments if an unrecognized action type is dispatched?

Reply:
removed the reducer in favor of external controlled state, this makes the component implementation and maintenance easier.
I will include usage examples in the stories.